### DISCUSSION TITLE
Select the GPU to use

### DISCUSSION THREAD

**Discussion by 7empest15** [lllyasviel/Fooocus#837] [Q&A]
Hello, I'd like to know how I could go about choosing which gpu to use. When I run the program, it uses my built-in gpu rather than my NVIDIA thanks

**Comment by kpagcha** [comment]
Same issue here.

**Comment by kienerj** [comment]
Same here. need to be able to select the GPU via CUDA_VISIBLE_DEVICES.

EDIT: just for clarity, you can go into entry_for_update.py (if that's how you start it) and add:

    os.environ['CUDA_VISIBLE_DEVICES'] = '1'

where the number is the gpu of your choice. However if you update this might get lost and it would be preferable to have this as a config option than hardcoded.

**Reply by divinets** [reply]
> ```
> os.environ['CUDA_VISIBLE_DEVICES'] = '1'
> ```

And for few AMD video please...
